paimon-core/src/main/java/org/apache/paimon/table/source/AbstractInnerTableScan.java

```diff
@@ -65,10 +65,14 @@ public CoreOptions options() {
     }
 
     protected StartingScanner createStartingScanner(boolean isStreaming) {
-        if (options.toConfiguration().get(CoreOptions.STREAMING_COMPACT)) {
+        if (options.toConfiguration().get(CoreOptions.STREAMING_COMPACT)
```

Comment:
extract `options.toConfiguration().get(CoreOptions.STREAMING_COMPACT)`  to a field